lib/error-matcher.js

```diff
@@ -25,12 +26,25 @@ module.exports = (function () {
   util.inherits(ErrorMatcher, EventEmitter);
 
   ErrorMatcher.prototype._gotoNext = function () {
+    if (0 === this.currentMatch.length) {
+      return;
+    }
+
     this._goto(this.currentMatch[0].id);
-    this.currentMatch.push(this.currentMatch.shift());
   };
 
   ErrorMatcher.prototype._goto = function (id) {
     var match = _.findWhere(this.currentMatch, { id: id });
+    if (!match) {
+      throw new Error('Can\'t find match with id ' + id);
+    }
```

Comment:
You're correct, that it schouldn't happen. It's more like defensive programming, if the other functions change in the future. But you are also correct, that it should emit an error.